ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,226 @@
+To obtain this, follow these steps:
 
+-   click 'Compute' in the Databricks menu, then click on your cluster's name
+-   in the address bar, copy the string of characters between `clusters` and `configuration` in the URL. This is your cluster ID.
 
 ------------------------------------------------------------------------
 
-# Pulling data into R studio from Databricks
+### Pulling data into RStudio from Databricks
 
 ------------------------------------------------------------------------
 
-Once you have set up an ODBC connection as detailed above, you can then use that connection to pull data directly from Databricks into R Studio. Charlotte recorded a video demonstrating two possible methods of how to do this. The recording is embedded below: 
+Now that you have established a connection between Databricks and your laptop, and between Databricks and RStudio, you can add code to your existing scripts to pull data into RStudio for analysis. If you have connected to other SQL databases before, this code will look quite familiar to you. 
+
+To access the data, we will make use of the `ODBC` package. You can find documentation about this package [on the Posit website](https://solutions.posit.co/connections/db/r-packages/odbc/).
+
+You can connect to your data in two ways, both of which are described below.
+
+---
+
+#### Connecting via the SQL Warehouse
+
+---
+
+This uses the `ODBC` package and can interact with tables in the Delta Lake. 
+
+Include the following code in your R Script:
+
+
+``` {r databricks_connect_sql, eval=FALSE}
+library(odbc)
+
+con <- DBI::dbConnect(
+    odbc::databricks(),
+    httpPath = Sys.getenv("DATABRICKS_SQL_WAREHOUSE_ID")
+)
+
+odbcListObjects(con)
+```
+
+---
+
+#### Connecting via a compute cluster
+
+---
+
+Compute clusters can interact with Volumes in Databricks, unlike the SQL Warehouse. For compute clusters, you need to use `sparklyr` rather than the `ODBC` package.
+
+Include the following code in your R Script:
+
+``` {r databricks_connect_cluster, eval=FALSE}
+library(tidyverse)
+library(sparklyr)
+library(dbplyr)
+
+sc <- spark_connect(
+  cluster_id = Sys.getenv("DATABRICKS_CLUSTER_ID"),
+  method = "databricks_connect"
+)
 
-<div align="center">
-<iframe src="https://educationgovuk.sharepoint.com/sites/lvewp00086/_layouts/15/embed.aspx?UniqueId=5fad039e-a763-40c8-8ea0-403aea712f4c&embed=%7B%22ust%22%3Atrue%2C%22hv%22%3A%22CopyEmbedCode%22%7D&referrer=StreamWebApp&referrerScenario=EmbedDialog.Create" width="640" height="360" frameborder="0" scrolling="no" allowfullscreen title="ADA_Rstudio_2.mp4"></iframe>
-</div>
+tbl(
+  sc,
+  I("catalog_10_gold.information_schema.catalogs")
+) %>%
+  head()
+```
 
-A template of all of the code used in the above video can be found in the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect).
+You can read CSV files from a Volume into a dataframe using the following code:
```

Comment:
We should say "Spark DataFrame" like in the [docs](https://rdrr.io/cran/sparklyr/man/spark_read_csv.html), since it's still processing everything on DataBricks, and to actually bring it in to a an R data.frame you'll need to run `collect()`.